Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM centos:centos7
+
+# ENV variables that need to be updated/supplied either from CLI or
+# docker-compose.yml
+ENV DATABASE_NAME=livedata
```

Comment:
These environment variables should not be hardcoded in the image but rather initialized at run-time when the container starts. Is there a reason why you put them in the Dockerfile?

Reply:
short version: 
Defining the required variable inside the dockerfile is the standard practice.

long version:
The `ENV` in a docker file is intended to define the env vars that can be later overwritten via docker-compose or CLI.
Therefore the actual values here does not matter as they will definitely be different when `live_data_server` image is integrated into other system.
If these values are not defined here, the image won't be able to recognize these are env variables, and there is a good chance that the runtime container will have some strange issue with these vars.

Reply:
> Defining the required variable inside the dockerfile is the standard practice.

The standard practice for sensitive information like the database admin account is to avoid encoding it the source, [like we're doing in web_reflectivity](https://code.ornl.gov/reflectometry/web_reflectivity/-/blob/next/cfg/docker-compose.localdev.yml#L49). You can place those variables in an `.envrc` file automatically read by command `direnv`, or you can manually export them in the terminal where you build your images.

Reply:
These are not sensitive information since they are already visible from the source code.
During the deployment, the actual Authentication information is stored locally, so there is no reason to make things extra cryptic.

@peterfpeterson please advise.

Reply:
Being a bit reductionist: having a password in the source today doesn't mean that it should stay there forever. 

If we were going to redeploy this for something other than the test environment, I think we'd need the ability to change where the container looks for the database and the login at runtime so there is only a configuration change (not a rebuild) to go from remote testing environment to production environment. I really don't have a strong preference, but if we leave them hard-coded with the test information now and we work on this repo "later" we'll need to remove them then.

Reply:
I do have a strong preference, though. I rather stick to the convention we have adopted regarding secrets so that we internalize this practice, instead of incurring in technical debt.

Reply:
I forgot to say that even if we're not setting a CI for this repo, we can store the secret environment variables as [CI secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) so that we won't forget their values and they'll be protected

Reply:
Github doesn't let you reveal secrets once they're set. :(

For the sake of consistency with other projects, let's not put the secrets in the repo and have them as CI/dev provided variables.

Reply:
These are __not__ secrets (even if we remove them from dockerfile, they are still visible from the settings.py, which means we can't hide them).
Also, defining them in dockerfile basically tell future devs that these are the environment vars you have to specify when spinning up a container based on the image.
Currently, live_data_server does not need to talk to the data base during the build, and the db action only happens

Reply:
> Github doesn't let you reveal secrets once they're set. :(

bummer, I thought the behavior was the same as GitLab's regarding environment variables.
I tried [adding a `DATABASE_USER`](https://github.com/neutrons/live_data_server/settings/secrets/actions) and you're right, once set there's no way to see its value.

Reply:
> These are **not** secrets (even if we remove them from dockerfile, they are still visible from the settings.py, which means we can't hide them).

[I'm looking at settings.py](https://github.com/neutrons/live_data_server/blob/REF292_containerization/live_data_server/live_data_server/settings.py#L92) and these variables are being referenced but I don't see their actual values hardcoded in the source. What do you mean they're visible?

Reply:
The values stored in the current dockerfiles are dummy ones, which will work locally.
You were suggesting removing the definition completely, which is what I am against as it is pointless:
- defining them in the dockerfile makes it easier for future devs to know that these are the necessary vars that need to be provided when pinning up a container
- the values of these env vars in the dockerfile are not important, but it allows one to quickly spin things up for checking.
- an important point I think is missing at the moment: when spinning up the image as a new container, docker __will update/overwrite__ these var with whatever is provided.